subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/UnavailableFileWatcherRegistryFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.watch.registry.impl;
+
+import org.gradle.internal.watch.registry.FileWatcherRegistry;
+import org.gradle.internal.watch.registry.FileWatcherRegistryFactory;
+
+public class UnavailableFileWatcherRegistryFactory implements FileWatcherRegistryFactory {
+    @Override
+    public FileWatcherRegistry createFileWatcherRegistry(FileWatcherRegistry.ChangeHandler handler) {
+        throw new UnsupportedOperationException("File watching is not available for this operating system");
+    }
```

Comment:
Can we use `Optional` here instead of a method that throws?

Reply:
No, because the `native` project needs to work on Java 6 :(